### DISCUSSION TITLE
[4.0] Radio button onchange form submit did not work in joomla filter bar

### DISCUSSION THREAD

**Discussion by HRIT-Florian** [joomla/joomla-cms#39200] [General]
### Steps to reproduce the issue
1.  Edit administrator\components\com_content\forms\filter_articles.xml
2. Add a radio button form field like 

```xml
<field
  name="text"
  type="radio"
  onchange="this.form.submit();"
  default="0"
>
	<option value="0">JNO</option>
	<option value="1">JYES</option>
</field>
```

3. Go to administrator/index.php?option=com_content&view=articles and try to toggle that radio button 

![radio_button_filter](https://user-images.githubusercontent.com/12803406/81854044-9acf5e80-955d-11ea-9c08-6faf9c0eac79.jpg)

### Expected result
onchange="this.form.submit();" should submit the form and the toggled radio button should be selected

### Actual result
nothing happens, js event did trigger

### System information (as much as possible)
Joomla! Version | Joomla! 4.0.0-beta1-dev Development [ Mañana ] 17-October-2019 20:21 GMT
Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:76.0) Gecko/20100101 Firefox/76.0

**Comment by SharkyKZ** [comment]
Looking at the code, `onchange` should be added to each individual option like this:

    <option value="0" onchange="this.form.submit();">JNO</option>

It's the same in J3. But different for `switcher` layout in J4.

**Comment by HRIT-Florian** [comment]
Yeah, did work by adding to option, but thats not clear and too much if you have a large list of options.
Anyway thanks for that info, but someone should fix it.

**Comment by SharkyKZ** [comment]
Well, there's an advantage to having this on options - each option can have a different onchange value.

**Comment by HRIT-Florian** [comment]
I don't think it's an advantage. It should be possible to set the handling onchange for all fields equally. If someone else wants to handle it different, he can write a separate field for it.

**Comment by joomdonation** [comment]
As @SharkyKZ pointed out, you can add onchange  to individual option. From documentation of our radio form field, onchange is not a supported attribute too. For the time being, it is not really an issue, I'm converting it to discussion and we can discuss further from there if needed.
